run/cisco2john.pl

```diff
@@ -44,188 +44,185 @@
-	);
+				 CISCO => "./0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz",
+				);
 
 sub usage {
-    my $dn = $0; $dn =~ s/(.*)\/cisco2john.pl/$1/;
-    print "Usage:\t$0 [cisco config file(s)] >>hashfile 2>>seed.txt\n";
-    print "\t${dn}/john -format:md5 -wordlist:seed.txt -rules hashfile\n\n";
-    exit 1;
+	my $dn = $0; $dn =~ s/(.*)\/cisco2john.pl/$1/;
+	print "Usage:\t$0 [cisco config file(s)] >>hashfile 2>>seed.txt\n";
+	print "\t${dn}/john -format:md5 -wordlist:seed.txt -rules hashfile\n\n";
+	exit 1;
 }
 
 sub Decode {
-  my $data = shift;
-  my $str = shift;  # Can supply custom Base64
-  my $pad="=";
-
-  my $type;
-  if ( defined $str && ! defined $Alphabets{$str} )  { $type = 'USER'; }
-  elsif ( defined $str && defined $Alphabets{$str} ) { $type = $str; }
-  #Try to guess
-  elsif ( $data =~ /[\.\/]/s && $data !~ /[+\/_-]/ ) { $type = 'CISCO'; }
-  else                                               { $type = 'CISCO'; } # Assume Standard Base64 if
-  if ( $type eq "USER" )                             { $Alphabets{'USER'} = $str; }
-
-  #strip non-base64 chars
-  my $estr;
-  if ( $Alphabets{$type} =~ /^(.{64})(.?)$/s ) { $str=$1; $estr=quotemeta($1); $pad=$2; } else { return undef; }
-  $data =~ s/[^$estr]//gs;
-
-  # Force Padding
-  $data .= $pad x (3-(((length($data)+3) % 4)));
-  $data=~s|(.)(.)(.?)(.?)|
-              chr(((index($str,$1)<<2)&252)+((index($str,$2)>>4)&3)).                      #six bits from first with two bits from the second
-              (($3 ne $pad)?chr(((index($str,$2)<<4)&240)+((index($str,$3)>>2)&15)):"").   #last 4 bits from second with four bits from third unless third is pad
-              (($4 ne $pad)?chr(((index($str,$3)<<6)&192)+((index($str,$4))&63)):"")       #last 2 bits from third with six bits from the forth unless forth is pad
-              |ge;
-  return $data;
+	my $data = shift;
+	my $str = shift;	# Can supply custom Base64
+	my $pad="=";
+
+	my $type;
+	if ( defined $str && ! defined $Alphabets{$str} ) { $type = 'USER'; }
+	elsif ( defined $str && defined $Alphabets{$str} ) { $type = $str; }
+	#Try to guess
+	elsif ( $data =~ /[\.\/]/s && $data !~ /[+\/_-]/ ) { $type = 'CISCO'; }
+	else { $type = 'CISCO'; } # Assume Standard Base64 if
+	if ( $type eq "USER" ) { $Alphabets{'USER'} = $str; }
+
+	#strip non-base64 chars
+	my $estr;
+	if ( $Alphabets{$type} =~ /^(.{64})(.?)$/s ) { $str=$1; $estr=quotemeta($1); $pad=$2; } else { return undef; }
+	$data =~ s/[^$estr]//gs;
+
+	# Force Padding
+	$data .= $pad x (3-(((length($data)+3) % 4)));
+	$data=~s|(.)(.)(.?)(.?)|
+	  chr(((index($str,$1)<<2)&252)+((index($str,$2)>>4)&3)).	#{six bits from first with two bits from the second
+	  (($3 ne $pad)?chr(((index($str,$2)<<4)&240)+((index($str,$3)>>2)&15)):"").	#{last 4 bits from second with four bits from third unless third is pad
+	  (($4 ne $pad)?chr(((index($str,$3)<<6)&192)+((index($str,$4))&63)):"")	#{last 2 bits from third with six bits from the forth unless forth is pad
```

Comment:
Somehow this adds `{` characters to the beginning of the 3 comments. Doesn't look like an indentation-only fix to me.

Reply:
Wow, can't believe I missed that. So when I fix that, what should I do with that licensing thing? Can I just add what he did as in "This software is released under the same terms as Perl itself. If you don't know what that means visit http://perl.com/."

Reply:
We can do that. Please use https://dev.perl.org/licenses/ for the URL. We can also add from there:

```
This is free software; you can redistribute it and/or modify it under the terms of either:

a) the GNU General Public License as published by the Free Software Foundation; either version 1, or (at your option) any later version, or

b) the Perl "Artistic License".
```